command/agent/template/template_test.go

```diff
@@ -116,6 +154,10 @@ func TestServerRun(t *testing.T) {
 					t.Fatalf("secret didn't match: %#v", secret)
 				}
 			}
+			if fileCount != len(templatesToRender) {
+				t.Fatalf("mismatch file to template: (%d) / (%d)", fileCount, len(templatesToRender))
+			}
+			cancelFunc()
```

Comment:
If we're setting `ExitAfterAuth` to true, we shouldn't need to call this (and removing it also tests that the exit after auth functionality works correctly since it shouldn't depend on context cancelation).

Reply:
Good catch, thanks! Removed in cb9558b29